### DISCUSSION TITLE
${_type} not replace in schema template other than sk/pk

### DISCUSSION THREAD

**Discussion by kinesra75** [sensedeep/dynamodb-onetable#218] [Q&A] [answered]
Hi,

I tried to add ${_type} placeholder on schema properties other than pk/sk (it works well on it) and I got "undefined" as result.

For exemple

    Product: {
      pk: { type: String, value: 'Enterprise#${enterpriseId}' },
      sk: { type: String, value: '${_type}#${id}' }, <=== WORKS
      id: { type: String, uuid: 'ulid' },
      enterpriseId: {
        type: String,
        required: true,
        custom: { faker: '{{$Enterprise.id}}' },
      },
      entEntity: {
        type: String,
        value: 'Enterprise#${enterpriseId}#${_type}', <=== NOT WORKING
        required: true,
      },
      status: { type: String, default: 'active', enum: ['active', 'inactive'] },
    },

As a result the field entEntity is : Enterprise#123#undefined

Edit : Also we can't use _type field in GSI, I got for exemple to add a field entity with the same value to be able to use an index like for exemple :

    gsi2: { hash: 'entity', sort: '_createdAt', follow: false },

Maybe I did something wrong ..

Thanks in advance.

Regards,
Sébastien Boucard

**Comment by mobsense** [comment, marked as answer]
Value template properties are hidden by default on return results. If you want them to be returned, set {hidden: true} in your params.

This default can be changed in the Table constructor params.

See this in the README:
```
Table Constructor Default
| hidden | `boolean` | Hide templated (value) attributes in Javascript properties. Default true. |

API params
| hidden | `boolean` | Set to true to omit the attribute in the returned Javascript results. Attributes with a "value" template defined will by hidden by default. Default to false. |

```

**Reply by mobsense** [reply]
And sorry, I don't understand:

>Edit : Also we can't use _type field in GSI, I got for exemple to add a field entity with the same value to be able to use an index like for exemple :
